### DISCUSSION TITLE
RSF to predict survival times

### DISCUSSION THREAD

**Discussion by ManuelaQuaresma** [kogalur/randomForestSRC#286] [General]
Dear Ishwaran,

I am using random (survival) forests for the first time. I have a dataset with individual level observed survival times, vital status and some covariates, including age. I would like to understand the possibility of predicting age and survival times for all individuals in my original dataset so that I can plot the observed versus predicted survival times (and ages). For age, I used the regression RF family to predict age and it worked well using the commands:

rf.obj <- rfsrc(age ~., data = data2)
rf.obj$predicted.oob

For the survival times, I tried to use the survival family command (note that I also tried a regression RF with time as the outcomes but I wanted to see if my prediction was improved with a RSF model): 

rsf.obj <- rfsrc(Surv(time, status) ~ ., data = data2)

But I am not able to predict the individual survival times for all individuals in my original data. My understanding so far is that for the RSF I can get the vector of mortality values (stored in the predicted vector), the CHF and survival function for specific times of interest. My question then is: is it possible to predict the survival times for each individual from an RSF (similar to my first RF model when I used age as the outcome). 

Many thanks in advance for any guidance you can give.

**Comment by ishwaran** [comment]
Yes it is possible.  It's stored in rsf.obj$surv.oob

Also see the [rsf vignette](https://www.randomforestsrc.org/articles/survival.html)

And this information is also described in the help file:

```
Values returned depend heavily on the family. 
In particular, predicted values from the forest (predicted and predicted.oob) are as follows:

**Also returned are matrices containing the CHF and survival function.
Each row corresponds to an individual's ensemble CHF or survival 
function evaluated at each time point in time.interest.**


```
